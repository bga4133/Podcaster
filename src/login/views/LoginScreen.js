import React from 'react'
import { Link } from 'react-router-dom'

export const LoginScreen = () => {
  return (
    <div className='loginScreen'>
      <Link  to="/" className='loginButton'>Go home</Link>
    </div>
  )
}
