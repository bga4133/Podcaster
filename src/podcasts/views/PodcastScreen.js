import React, { useEffect, useCallback, useState, useMemo } from 'react'
import { Loader } from '../../components/loader/Loader';
import { Podcastlist } from './podcastList/PodcastList';
import { SearchPodcast } from './searchPodcast/SearchPodcast';

export const PodcastScreen = () => {
    const [IsLoading, setIsLoading] = useState(false);
    const [PodcastResponse, setPodcastResponse] = useState([]);
    const [query, setQuery] = useState('')

    const getPodcasts = async () => {
        setIsLoading(true);
        try {
            const response = await fetch ("https://itunes.apple.com/us/rss/toppodcasts/limit=100/genre=1310/json");
            const { feed } = await response.json();
            setPodcastResponse(feed.entry)
            setIsLoading(false);
        } catch (e) {
            setIsLoading(false);
            throw new Error(e);
        }
    }
    
    let podcast = useMemo(() => PodcastResponse, [PodcastResponse]);
    const filterSearch = podcast.map((item1) => item1).
    filter((item) => item.title.label.toLowerCase().includes(query) || item['im:artist'].label.toLowerCase().includes(query))
    const responseLength = podcast.length;
    useEffect(() => {
      getPodcasts();
    }, [])

  return (
    <div>
      <SearchPodcast setQuery={setQuery} responseLength={responseLength} />
      {IsLoading && <Loader />}
      <div  className='podcastScreen' data-testid="imgId">
      {filterSearch.map((item) => (
        <>
        <Podcastlist key={item.id.attributes['im:id']} item={item} />
        </>
      )) }
      </div>
      {PodcastResponse == null && <p>Error...</p>}
    </div>
  )
}


