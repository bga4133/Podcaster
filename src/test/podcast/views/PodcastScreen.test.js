import { render, screen, waitFor } from "@testing-library/react"
import { PodcastScreen } from "../../../podcasts/views/PodcastScreen"

describe('<PodcastScreen /> test', () => {
    const { container } = render(<PodcastScreen />);
    it("render the component", async () => {
        expect(container).toMatchSnapshot();
    })
    it("shouldn't show the div with the class podcastScreen if data is no loadet", async () => {
        const containerDiv = container.querySelector('podcastScreen');
        expect(containerDiv).toBeFalsy();
    });
    it("should show the data after load the page ", async () => {
        const { container } = render(<PodcastScreen  />);
        const containerDiv = container.querySelector("container");
        await waitFor(
          () => {
            expect(screen.getByTestId("imgId")).toBeInTheDocument();
          },
          { timeout: 4000 }
        );
      });
    
});