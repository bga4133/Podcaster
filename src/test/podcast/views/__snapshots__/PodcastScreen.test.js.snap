// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PodcastScreen /> test render the component 1`] = `
<div>
  <div>
    <div
      class="searchContainer"
    >
      <p
        class="responseLength"
      >
        0
      </p>
      <input
        class="search"
        placeholder="search"
        type="text"
      />
    </div>
    <div
      class="lds-dual-ring"
    />
    <div
      class="podcastScreen"
      data-testid="imgId"
    />
  </div>
</div>
`;
